libres/lib/res_util/matrix_lapack.cpp

```diff
@@ -422,36 +380,24 @@ void matrix_dorgqr(matrix_type *A, double *tau,
     int lda = matrix_get_column_stride(A);
     int m = matrix_get_rows(A);
     int n = matrix_get_columns(A);
-    double *work = (double *)util_malloc(sizeof *work);
+    std::vector<double> work(1);
     int worksize;
     int info;
 
     /* Determine optimal worksize. */
     worksize = -1;
-    dorgqr_(&m, &n, &num_reflectors, matrix_get_data(A), &lda, tau, work,
+    dorgqr_(&m, &n, &num_reflectors, matrix_get_data(A), &lda, tau, work.data(),
             &worksize, &info);
     if (info != 0)
         util_abort("%s: dorgqf routine failed with info:%d \n", __func__, info);
-    worksize = (int)work[0];
-    {
-        double *tmp = (double *)realloc(work, sizeof *work * worksize);
-        if (tmp == NULL) {
-            /*
-         OK - we could not get the optimal worksize,
-         try again with the minimum.
-      */
-            worksize = n;
-            work = (double *)util_realloc(work, sizeof *work * worksize);
-        } else
-            work = tmp; /* The request for optimal worksize succeeded */
-    }
+    worksize = static_cast<int>(work[0]);
```

Comment:
Unnecessary cast.

Reply:
This is a `double`&rightarrow; `int` conversion - I thought it was nice to make it explicit; but I am happy to remove the cast if you prefer.